pennylane_lightning_kokkos/src/algorithms/MeasuresKokkos.hpp

```diff
@@ -0,0 +1,585 @@
+#pragma once
+
+#include <Kokkos_Core.hpp>
+#include <Kokkos_Random.hpp>
+
+#include "ExpValFunctors.hpp"
+#include "LinearAlgebra.hpp"
+#include "MeasuresFunctors.hpp"
+#include "ObservablesKokkos.hpp"
+#include "StateVectorKokkos.hpp"
+
+namespace Pennylane::Algorithms {
+
+template <class Precision> class MeasuresKokkos {
+
+  private:
+    using KokkosExecSpace = Kokkos::DefaultExecutionSpace;
+    using KokkosVector = Kokkos::View<Kokkos::complex<Precision> *>;
+    using KokkosSizeTVector = Kokkos::View<size_t *>;
+    using UnmanagedSizeTHostView =
+        Kokkos::View<size_t *, Kokkos::HostSpace,
+                     Kokkos::MemoryTraits<Kokkos::Unmanaged>>;
+    using UnmanagedPrecisionHostView =
+        Kokkos::View<Precision *, Kokkos::HostSpace,
+                     Kokkos::MemoryTraits<Kokkos::Unmanaged>>;
+    using UnmanagedConstComplexHostView =
+        Kokkos::View<const Kokkos::complex<Precision> *, Kokkos::HostSpace,
+                     Kokkos::MemoryTraits<Kokkos::Unmanaged>>;
+    using UnmanagedConstSizeTHostView =
+        Kokkos::View<const size_t *, Kokkos::HostSpace,
+                     Kokkos::MemoryTraits<Kokkos::Unmanaged>>;
+
+    const StateVectorKokkos<Precision> &original_sv;
+
+    using ExpValFunc = std::function<Precision(const std::vector<size_t> &,
+                                               const std::vector<Precision> &)>;
+    using ExpValMap = std::unordered_map<std::string, ExpValFunc>;
+    ExpValMap expval_funcs_;
+
+  public:
+    explicit MeasuresKokkos(const StateVectorKokkos<Precision> &state_vector)
+        : original_sv{state_vector} {
+        expval_funcs_["Identity"] = [&](auto &&wires, auto &&params) {
+            return getExpectationValueIdentity(
+                std::forward<decltype(wires)>(wires),
+                std::forward<decltype(params)>(params));
+        };
+
+        expval_funcs_["PauliX"] = [&](auto &&wires, auto &&params) {
+            return getExpectationValuePauliX(
+                std::forward<decltype(wires)>(wires),
+                std::forward<decltype(params)>(params));
+        };
+
+        expval_funcs_["PauliY"] = [&](auto &&wires, auto &&params) {
+            return getExpectationValuePauliY(
+                std::forward<decltype(wires)>(wires),
+                std::forward<decltype(params)>(params));
+        };
+
+        expval_funcs_["PauliZ"] = [&](auto &&wires, auto &&params) {
+            return getExpectationValuePauliZ(
+                std::forward<decltype(wires)>(wires),
+                std::forward<decltype(params)>(params));
+        };
+
+        expval_funcs_["Hadamard"] = [&](auto &&wires, auto &&params) {
+            return getExpectationValueHadamard(
+                std::forward<decltype(wires)>(wires),
+                std::forward<decltype(params)>(params));
+        };
+    };
+
+    /**
+     * @brief Calculate the expectation value of an observable
```

Comment:
```suggestion
     * @brief Calculate the expectation value of a named observable
```

Since we expect the string to register with a given named obs type, we should clarify in the brief.

Reply:
Just copied it from StateVectorKokkos! Fixed. Thanks!